src/System.Windows.Forms/src/System/Windows/Forms/TabControl.cs

```diff
@@ -1202,7 +1202,7 @@ internal void Insert(int index, TabPage tabPage)
         /// </summary>
         private void InsertItem(int index, TabPage tabPage)
         {
-            if (index < 0 || ((_tabPages is not null) && index > _tabPageCount))
+            if (index < 0 || (index > _tabPageCount))
```

Comment:
This change is done to make sure that you can't insert a tabpage with index > 0 when there are no tabpages in the tabcontrol.
So now it throws an error when a tabpage gets inserted with a too high index, even when there are no tabpages yet.
But this changes the behaviour of the tabcontrol, so we should discuss this.

Reply:
`_tabPages is not null` was here to ensure `Insert(int index, TabPage tabPage)` was called, which not allocated the backing array, but also created the first page. Now if we try to insert a page into a new instance of `TabControl` `_tabPageCount==0` and hence we stumble on `index > _tabPageCount` (e.g. `0 > 0`). 
Is this what you're referring to?

If so, would this work?
```suggestion
            if (index < 0 || (index != 0 && index > _tabPageCount))
```